server/Arcadia.Assistant.CSP/ArcadiaHealthChecker.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace Arcadia.Assistant.CSP
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    using Akka.Actor;
+    using Health.Abstractions;
+    using Microsoft.EntityFrameworkCore;
+    using Model;
+    using Vacations;
+
+    public class ArcadiaHealthChecker : HealthChecker, ILogReceive
+    {
+        private const string HealthStateName1C = "1C";
+        private const string HealthStateNameDatabase = "Database";
+
+        private readonly Func<ArcadiaCspContext> contextFactory;
+        private readonly Remote1CConfiguration bkConfiguration;
+
+        public ArcadiaHealthChecker(Func<ArcadiaCspContext> contextFactory, Remote1CConfiguration bkConfiguration)
+        {
+            this.contextFactory = contextFactory;
+            this.bkConfiguration = bkConfiguration;
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case HealthCheckMessage _:
+                    var checkDatabaseResult = this.CheckDatabase();
+
+                    this.Check1C().PipeTo(
+                        this.Sender,
+                        success: check1CResult =>
+                        {
+                            var healthResult = new Dictionary<string, bool>
+                            {
+                                { HealthStateName1C, check1CResult },
+                                { HealthStateNameDatabase, checkDatabaseResult }
+                            };
+                            return new HealthCheckMessageResponse(healthResult);
+                        });
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private async Task<bool> Check1C()
+        {
+            try
+            {
+                var uriBuilder = new UriBuilder(this.bkConfiguration.Url);
+                var client = new EnterpriseV8(uriBuilder.Uri)
```

Comment:
why do that when you can just store success/failure of the last operation?

Reply:
> sponkrashin 2 days ago  Member
It is not really proper way, because it violates SRP rule. Actor is starting to do not only what it should do, but also some additional work. But for now let it be so.

> sponkrashin 2 days ago  Member
Also minus is that currently period of refresh is once in 10 minutes. It is not frequent enough.

> sponkrashin 2 days ago  Member
Kept as is - all checks are executed inside health actor. Reasons were listed above.

Reply:
Done.